packages/layout/src/tree-layout/layout-definition/helper/index.js

```diff
@@ -0,0 +1,290 @@
+import { NODE_PREFIX } from '../../constants/defaults';
+import DefinitionModel from '../definition-manager/definitionModel';
+import DummyComponent from '../../layout-component/dummy-component';
+/**
+ * Compares two strings in lowercase
+ *
+ * @export
+ * @param {string} value the value to be compared
+ * @param {string} compareTo whom the value would be compared
+ * @return {boolean} true if values are equal
+ */
+export function isEqual (value, compareTo) {
+    if (typeof value !== 'string' || typeof compareTo !== 'string') {
+        throw new TypeError('value and compareTo must be string');
+    }
+    return value.toLowerCase() === compareTo.toLowerCase();
+}
+
+export const getNodeId = (() => {
+    let _uid = 0;
+    return () => `${NODE_PREFIX}-${++_uid}`;
+})();
+
+export function yExtraSpace (node) {
+    let smallestHeight = 0;
+    if (node.getCutType() === 'v') {
+        smallestHeight = smallestExtraHeightHorizontally(node); // eslint-disable-line no-use-before-define
+    } else if (node.getCutType() === 'h') {
+        node.children().forEach((child) => {
+            smallestHeight += yExtraSpace(child);
+        });
+    } else if (node.model().host() && node.model().host().getLogicalSpace) {
+        const containerHeight = node.boundBox().height;
+        const hostHeight = node.model().host().getLogicalSpace().height;
+
+        smallestHeight = containerHeight - hostHeight;
+        if (smallestHeight < 0) {
+            smallestHeight = 0;
+        }
+    } else {
+        smallestHeight = 0;
+    }
+    return smallestHeight;
+}
+
+export function smallestExtraHeightHorizontally (node) {
+    let smallestHeight = Number.MAX_SAFE_INTEGER;
+    node.children().forEach((child) => {
+        const h = yExtraSpace(child);
+        if (h < smallestHeight) {
+            smallestHeight = h;
+        }
+    });
+    return smallestHeight;
+}
+
+export function xExtraSpace (node) {
+    let smallestWidth = 0;
+    if (node.getCutType() === 'h') {
+        smallestWidth = smallestExtraWidthVertically(node); // eslint-disable-line no-use-before-define
+    } else if (node.getCutType() === 'v') {
+        node.children().forEach((child) => {
```

Comment:
This function and yExtraSpace have the same function definition with some changes in assigning width/ height.
Use either a common function or better yet, use strategy pattern.